yt/fields/field_info_container.py

```diff
@@ -448,16 +449,15 @@ def find_dependencies(self, loaded):
         return loaded, unavailable
 
     def add_output_field(self, name, sampling_type, **kwargs):
+        if name[1] == "density":
+            if name in self:
+                # this should not happen, but it does
```

Comment:
I am still very much confused after reading this comment. Why is this happening?

Reply:
TBH I don't know. I assume some frontends must be incorrectly setting the `("gas", "density")` field more than once

Reply:
Should I add a comment to clarify that I really mean `# TODO: remove this check and fix frontends that try to register a field twice` ?

Reply:
That's fine.